crates/matrix-sdk/src/sliding_sync.rs

```diff
@@ -878,17 +910,24 @@ impl SlidingSyncView {
     }
 
     #[tracing::instrument(skip(self, ops))]
-    fn room_ops(&self, ops: &Vec<v4::SyncOp>) -> anyhow::Result<()> {
+    fn room_ops(&self, ops: &Vec<v4::SyncOp>) -> Result<(), Error> {
         let mut rooms_list = self.rooms_list.lock_mut();
         let _rooms_map = self.rooms.lock_mut();
         for op in ops {
             match &op.op {
                 v4::SlidingOp::Sync => {
                     let start: u32 = op
                         .range
-                        .context("`range` must be present for Sync and Update operation")?
+                        .ok_or_else(|| {
+                            Error::BadResponse(
+                                "`range` must be present for Sync and Update operation".to_owned(),
+                            )
+                        })?
                         .0
-                        .try_into()?;
+                        .try_into()
+                        .map_err(|e| {
+                            Error::BadResponse(format!("`range` not a valid int: {:}", e))
```

Comment:
In addition to what clippy says, the `:` here is also a no-op AFAICT. So I think this should be

```suggestion
                            Error::BadResponse(format!("`range` not a valid int: {e}"))
```